networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -395,3 +457,76 @@ def _set_snat_pools_from_hosting_info(self, ri, gw_port, is_delete):
+        # is not aware of tenant routers with ext network assigned. Thus,
+        # default route must be handled per tenant router.
+        ex_gw_ip = ext_gw_port['subnets'][0]['gateway_ip']
+        sub_interface = self._get_interface_name_from_hosting_port(ext_gw_port)
+        vlan_id = self._get_interface_vlan_from_hosting_port(ext_gw_port)
+        if (self._fullsync and
+                int(vlan_id) in self._existing_cfg_dict['interfaces']):
+            LOG.debug("Sub-interface already exists, don't create "
+                      "interface")
+        else:
+            LOG.debug("Adding IPv4 external network port: %(port)s for tenant "
+                      "router %(r_id)s", {'port': ext_gw_port['id'],
+                                          'r_id': ri.id})
+            if ri.router['admin_state_up'] is False:
+                self._create_sub_interface_disable_only(sub_interface)
+            else:
+                if ext_gw_port['admin_state_up'] is False:
+                    self._create_sub_interface_disable_only(sub_interface)
+                else:
+                    self._create_sub_interface_enable_only(sub_interface)
+        if ex_gw_ip:
+            # Set default route via this network's gateway ip
+            if self._is_port_v6(ext_gw_port):
+                self._add_default_route_v6(ri, ex_gw_ip, ext_gw_port)
+            else:
+                self._set_nat_pool(ri, ext_gw_port, False)
+                self._add_default_route(ri, ext_gw_port)
+
+        # Global conifguration parameters are passed using the global
+        # router. Check to see if there is any global config, and if it's
+        # not already configured, add it in
+        g_configs = ext_gw_port['hosting_info'].get('global_config')
+        if g_configs and isinstance(g_configs, list):
+            self._set_global_config(ri, ext_gw_port, g_configs)
+
+    def _replace_template_vars(self, ri, port, config):
+        conf_str = config
+        for key in self._template_dict.keys():
+            if key in conf_str:
+                conf_str = self._template_dict[key](ri, port, conf_str)
+        return conf_str
+
+    def _config_cmd(self, ri, port, asr_str, g_config, remove=False):
+        if isinstance(g_config, list):
+            for config in g_config:
+                self._config_cmd(ri, port, asr_str, config, remove)
+        else:
+            if remove:
+                snippet = snippets.REMOVE_GLOBAL_CONFIG
+            else:
+                snippet = snippets.SET_GLOBAL_CONFIG
+            asr_str.append(snippet %
+                (self._replace_template_vars(ri, port, g_config)))
+
+    def _set_global_config(self, ri, port, g_configs):
+        for g_config in g_configs:
+            asr_str = [snippets.GLOBAL_CONFIG_PREFIX]
+            self._config_cmd(ri, port, asr_str, g_config)
+            asr_str.append(snippets.GLOBAL_CONFIG_POSTFIX)
+            self._edit_running_config(''.join(asr_str), 'SET_GLOBAL_CONFIG')
```

Comment:
It's an array of strings. It's faster/more efficient to do it this way than doing + with strings (according to google :-) ).